### ISSUE TITLE
[BUG] Kotlin client: wrong code for generated enums

### ISSUE THREAD

**Issue by kroegerama** [OpenAPITools/openapi-generator#2979]
##### Description

kotlin-client generates non-compilable code

##### openapi-generator version

4.0.0

##### OpenAPI declaration file content or url

```
gender:
        format: int32
        description: "* Geschlecht (0 = neutrale Anrede, 1 = Frau, 2 = Mann, 3 = Familie)\r\n* Gender (0 = neutral salutation, 1 = Female, 2 = Male, 3 = Family)"
        enum:
          - 0
          - 1
          - 2
          - 3
        type: integer
```

##### Command line used for generation

`[...] -g kotlin [...]`

##### Generated code

```
/**
    * * Familienstand (0 = single, 1 = verheiratet, 2 = geschieden, 3 = getrennt lebend, 4 = verwitwet)  * Marital status (0 = single, 1 = married, 2 = divorced, 3 = seperated, 4 = widowed)
    * Values: _0,_1,_2,_3,_4
    */
    enum class MaritalStatus(val value: kotlin.Int){
    
        @Json(name = 0)
        _0(0),
    
        @Json(name = 1)
        _1(1),
    
        @Json(name = 2)
        _2(2),
    
        @Json(name = 3)
        _3(3),
    
        @Json(name = 4)
        _4(4);
    
    }
```

`@Json(name` <- has to be a String.

**Comment by Zomzog** [comment]
I've tried it with master and 4.0.0 and got :
```
enum class Gender(val value: kotlin.Int){

    @Json(name = "0")
    _0(0),

    @Json(name = "1")
    _1(1),

    @Json(name = "2")
    _2(2),

    @Json(name = "3")
    _3(3);

}
```

Could you give me the complete command you use?

**Comment by kroegerama** [comment]
Sure. I used this jar: `http://central.maven.org/maven2/org/openapitools/openapi-generator-cli/4.0.0/openapi-generator-cli-4.0.0.jar`

The command was: `java -jar ./openapi-generator-cli-4.0.0.jar generate -g kotlin -c config.json -i swagger.yaml -o generated`

The config file:
```
{
  "packageName": "my.project.api",
  "groupId": "my.project",
  "dateLibrary": "threetenbp",
  "collectionType": "list"
}
```

The swagger file is `swagger: '2.0'`

**Comment by edawn** [comment]
I can confirm this issue still exists with openapi-generator version 4.0.3

**Comment by falejand** [comment]
> I can confirm this issue still exists with openapi-generator version 4.1.0

+1
Happening on 4.1.0 and 4.2.0

**Comment by NicolaeGP** [comment]
> I can confirm this issue still exists with openapi-generator version 4.1.0

+1
Happening on 4.2.2

**Comment by kroegerama** [comment]
@edawn @falejand @NicolaeGP 
You may want to try my new project:
https://github.com/kroegerama/openapi-kgen

It's a new generator, written entirely from scratch. A generated example can be found here:
https://github.com/kroegerama/openapi-kgen-generated

**Comment by ackintosh** [comment]
Ahh, I've confirmed the spec below reproduces the issue:

```yaml
openapi: 3.0.0
servers:
  - url: 'http://petstore.swagger.io/v2'
info:
  description: test
  version: 1.0.0
  title: OpenAPI Petstore
  license:
    name: Apache-2.0
    url: 'https://www.apache.org/licenses/LICENSE-2.0.html'
paths:
  /pet:
    get:
      summary: Finds Pets by status
      description: Multiple status values can be provided with comma separated strings
      operationId: findPetsByStatus
      responses:
        '200':
          description: successful operation
          content:
            application/json:
              schema:
                type: string
components:
  schemas:
    foo:
      type: object
      properties:
        gender:
          format: int32
          description: "test"
          enum:
            - 0
            - 1
            - 2
            - 3
          type: integer
```


The non-compilable code is generated from:

https://github.com/OpenAPITools/openapi-generator/blob/26c9c64634d30b9003da0758c230d5393607d89e/modules/openapi-generator/src/main/resources/kotlin-client/data_class.mustache#L63

**Comment by jimschubert** [comment]
Fixed in #5205 5205
